ExampleMod/Common/Systems/SystemUI.cs

```diff
@@ -0,0 +1,56 @@
+﻿using ExampleMod.Common.UI;
+using Microsoft.Xna.Framework;
+using System.Collections.Generic;
+using Terraria;
+using Terraria.ModLoader;
+using Terraria.UI;
+
+namespace ExampleMod.Common.Systems
+{
+	public class SystemUI: ModSystem
+	{
+		private UserInterface exampleCoinUserInterface;
+		internal ExampleCoinsUI exampleCoinsUI;
+
+		public override void Load() {
+			// Create custom interface wich can swap between difrent UIStates
+			exampleCoinUserInterface = new UserInterface();
+			// Creating custom UIState
+			exampleCoinsUI = new ExampleCoinsUI();
+
+			// Activate calls Initialize() on the UIState if not initialized, then calls OnActivate and then calls Activate on every child element
+			exampleCoinsUI.Activate();
+		}
+
+		public override void UpdateUI(GameTime gameTime) {
+			// Here will call .Update on custom UI and propagate it to its state and underlying elements
+			if (exampleCoinUserInterface?.CurrentState != null) 
+				exampleCoinUserInterface?.Update(gameTime);
+		}
+
+		// This two methods will set the state to custom UI, causing it to show or hide(if we pass null into the method call)
+		public void ShowMyUI() {
+			exampleCoinUserInterface?.SetState(exampleCoinsUI);
+		}
+		public void HideMyUI() {
```

Comment:
Add a newline, and possibly move both methods to the top of the class, before the first override
```suggestion
		}
		
		public void HideMyUI() {
```